src/braket/aws/aws_quantum_task.py

```diff
@@ -272,6 +272,8 @@ def result(self) -> Union[GateModelQuantumTaskResult, AnnealingQuantumTaskResult
         """
         if self._result or self._status(True) in self.NO_RESULT_TERMINAL_STATES:
             return self._result
+        if self._status(True) in self.RESULTS_READY_STATES:
```

Comment:
Note this has one of the issues being fixed in https://github.com/aws/amazon-braket-sdk-python/pull/234/files .